.github/workflows/coverage.yml

```
@@ -0,0 +1,85 @@
+name: Coverage
+
+on:
+  push:
+    branches:
+    - main
+  pull_request:
+    types: [opened, synchronize, reopened, ready_for_review]
+
+jobs:
+  BuildAndTest:
+    runs-on: ubuntu-latest
+
+    timeout-minutes: 60
+    steps:
+    - name: Check out code
+      uses: actions/checkout@v2
+
+    - name: Set up CCache
+      id: setup-ccache
+      run: |
+        sudo apt-get install ccache
+        ccache --max-size=2G
+
+        # Determine common base between main and HEAD to use as cache key.
+        if [ "${{ github.event_name }}" == "push" ]; then
+          BASE=$( git rev-parse HEAD )
+        else
+          BASE=$( git cat-file -p HEAD | awk '/parent/ { print $2; exit }' )
+        fi
+        echo "::set-output name=key::${BASE}"
+
+    - name: CCache
+      uses: actions/cache@v2
+      with:
+        key: ccache-${{ steps.setup-ccache.outputs.key }}
+        restore-keys: |
+          ccache-
+        path: ~/.ccache/**
+
+    - name: Install dependencies
+      run: |
+        sudo apt-add-repository --yes ppa:mir-team/dev
+        sudo apt install --no-install-recommends \
+          devscripts \
+          dmz-cursor-theme \
+          equivs \
+          gcovr \
+          lcov \
+          ninja-build
+
+        mk-build-deps --root-cmd sudo --install
+
+    - name: Configure
+      run: >
+        cmake
+        -DCMAKE_BUILD_TYPE=Coverage
```

As discussed in #2395
```suggestion
        -DCMAKE_BUILD_TYPE=Coverage
        -DMIR_RUN_PERFORMANCE_TESTS=ON        
```